griptape-ai/griptape
# Arxiv Tool Implementation ⛓️

### Title: Add Arxiv Tool Implementation

#### Description

This PR adds a new tool for searching scientific articles on Arxiv.org. The tool uses the Griptape framework and follows best practices for implementation and error handling.

#### Changes

- Adds `ArxivAPIWrapper` class to handle interactions with the Arxiv API.
- Adds `ArxivSearch` class, a Griptape `BaseTool`, to perform searches and return results.
- Adds schema validation for search parameters.

#### Files Changed

- `tool.py`
- `__init__.py`
- `manifest.yml`

#### How to Test

1. Install all dependencies listed: `arxiv & fitz`.
2. Run create a `.py` to ensure the Arxiv Tool is correctly fetching data.
```import os
from griptape.drivers import OpenAiChatPromptDriver
from griptape.rules import Rule
from griptape.structures import Agent
# Make sure DDGSearch is imported from where you've defined it.
from griptape.tools.arxiv import ArxivSearch
from dotenv import load_dotenv

load_dotenv()

# Create an agent with the DDGSearch tool
agent = Agent(
    prompt_driver=OpenAiChatPromptDriver(
        api_key=os.getenv("OPENAI_API_KEY"),
        model="gpt-3.5-turbo-16k",
        temperature=0,
    ),
    tools=[ArxivSearch()],
    rules=[
        Rule(value="Use the ArxivSearch tool to search & explain any scientific document. Use targeted search to get accurate information."),
        Rule(value="You'll give a detailed information about the topic, make it big but simple to understand."),
        Rule(value="Give roper citation, to give credit to authors."),
        Rule(value="You are a pirate, use pirate accent to give the output.")

    ],
)

# Run a search query to find information about what griptape is
result = agent.run(
    "Explain: Marijuana usage in cancer",
)
print(result.output)
```

3. Ensure all existing tests pass, and new tests cover the added functionality.


### Summary:

- A new tool for searching scientific articles on Arxiv.org has been added.
- The tool is implemented as a Griptape `BaseTool` and uses an `ArxivAPIWrapper` class for API interactions.
- Schema validation and error handling are included.
- Reviewers are asked to focus on code quality, functionality, and documentation.
